lib/params.go

```diff
@@ -99,3 +101,36 @@ func (p ListParams) Page() util.Page {
 	number = p.Offset/size + 1
 	return util.NewPage(number, size)
 }
+
+// GetParams defines parameters for looking up the head or body of a dataset
+type GetParams struct {
+	// Refstr to get, representing a dataset ref to be parsed
+	Refstr   string
+	Selector string
+
+	// read from a filesystem link instead of stored version
+	Format       string
+	FormatConfig dataset.FormatConfig
+
+	Limit, Offset int
+	All           bool
+
+	// outfile is a filename to save the dataset to
+	Outfile string
+	// whether to generate a filename from the dataset name instead
+	GenFilename bool
+	Remote      string
+
+	// Proxy identifies whether a call has been proxied from another instance
+	Proxy bool
+}
+
+// GetResult combines data with it's hashed path
+type GetResult struct {
+	Ref       *dsref.Ref       `json:"ref"`
+	Dataset   *dataset.Dataset `json:"data"`
+	Bytes     []byte           `json:"bytes"`
+	Message   string           `json:"message"`
+	FSIPath   string           `json:"fsipath"`
+	Published bool             `json:"published"`
+}
```

Comment:
My take was that since we're going to enforce an API wide move to directly work on these params, it's much more readable having it in a central place. I'll undo it.

Reply:
Yeah sorry. I totally agree it's a bit of a toss-up. Personally having written a bunch of these I need the struct def close to it's use, and instead often use [godoc](https://pkg.go.dev/github.com/qri-io/qri/lib#pkg-types) to audit across struct definitions.

In practice reusing input param structs has been a bad idea, but aligning field names is _super_ important. reusing output params (or ideally, returning common stuff like *dataset.Dataset) has been very good though.